sdk/ai/ai-projects/src/agents/runs.ts

```diff
@@ -1,31 +1,43 @@
 // Copyright (c) Microsoft Corporation.
 // Licensed under the MIT License.
 
-import { Client, createRestError } from "@azure-rest/core-client";
-import { CancelRunParameters, CreateRunParameters, CreateThreadAndRunParameters, GetRunParameters, ListRunsParameters, SubmitToolOutputsToRunParameters, UpdateRunParameters } from "../generated/src/parameters.js";
-import { OpenAIPageableListOfThreadRunOutput, ThreadRunOutput } from "../generated/src/outputModels.js";
+import type { Client } from "@azure-rest/core-client";
+import { createRestError, operationOptionsToRequestParameters } from "@azure-rest/core-client";
+import type * as GeneratedParameters from "../generated/src/parameters.js";
+import type { OpenAIPageableListOfThreadRunOutput, ThreadRunOutput } from "../generated/src/outputModels.js";
 import { validateLimit, validateMessages, validateMetadata, validateOrder, validateRunId, validateThreadId, validateTools, validateTruncationStrategy } from "./inputValidations.js";
 import { TracingUtility } from "../tracing.js";
 import { traceEndCreateOrUpdateRun, traceEndSubmitToolOutputsToRun, traceStartCreateRun, traceStartCreateThreadAndRun, traceStartSubmitToolOutputsToRun } from "./runTrace.js";
 import { traceStartAgentGeneric } from "./traceUtility.js";
 import { createRunStreaming, createThreadAndRunStreaming, submitToolOutputsToRunStreaming } from "./streaming.js";
-import { AgentEventMessageStream } from "./streamingModels.js";
-import { AgentRunResponse } from "./customModels.js";
+import type { AgentEventMessageStream } from "./streamingModels.js";
+import type { AgentRunResponse, CreateRunOptionalParams } from "./customModels.js";
+import { convertCreateRunOptions } from "../customization/convertModelsToWrite.js";
+import { convertThreadRunOutput } from "../customization/convertOutputModelsFromWire.js";
 
 const expectedStatuses = ["200"];
 
 /** Creates and starts a new run of the specified thread using the specified agent. */
 export function createRun(
   context: Client,
   threadId: string,
-  options: CreateRunParameters,
+  assistantId: string,
+  options: CreateRunOptionalParams,
```

Comment:
do we need options to be required here? If it's just for convenience you can make it have a default value of the empty object

Reply:
Its for convenience while converting, operationOptionsToRequestParameters expects the options be not emtpty